### PR TITLE
Fix TECS bug preventing flare for subsequent landings

### PR DESCRIPTION
**Pull request by priseborough**
This bug was reported by https://github.com/ArduPilot/ardupilot/pull/22191#issuecomment-1325428328 

The failure to reset the counter after exiting flare was preventing subsequent flares being performed unless the vehicle was power cycled after landing.

The  flare control no longer requires a counter to schedule the pitch limits, so this has been replaced with a boolean as suggested by https://github.com/ArduPilot/ardupilot/pull/22191#issuecomment-1325443278

### CODE REVIEW

File: libraries/AP_TECS/AP_TECS.cpp [hunk 1]
```diff
@@ -1260,13 +1262,6 @@ void AP_TECS::update_pitch_throttle(int32_t hgt_dem_cm,
         _PITCHminf = MAX(_land_pitch_min, _PITCHminf);
     }
 
-    if (_landing.is_flaring()) {
```
Comment (IamPete1) [inline libraries/AP_TECS/AP_TECS.cpp:1263]: Its not clear to me at first glace why this has been removed?

Comment (priseborough) [reply libraries/AP_TECS/AP_TECS.cpp:1263]: The flare specific setting of _PITCHmaxf  is already done at line 1237

Comment (priseborough) [reply libraries/AP_TECS/AP_TECS.cpp:1263]: I should have split this out into a separate commit.

Comment (magicrub) [reply libraries/AP_TECS/AP_TECS.cpp:1263]: basically, its just a consolidation using the above code. Same goes with the extra _landing.is_on_approach() call that could just get rolled into the same section above.

Comment (priseborough) [reply libraries/AP_TECS/AP_TECS.cpp:1263]: I've split this change into a separate commit

Comment (IamPete1) [reply libraries/AP_TECS/AP_TECS.cpp:1263]: Ah, just outside the little window github shows us ;).

**Review by IamPete1 — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**